### PR TITLE
Use the image digest in openshift-install, write a release summary to the output, and allow simple signing

### PR DESCRIPTION
**Pull request by smarterclayton**
Allows a tool publisher to guarantee the root of trust from a given release via install.  Ensures the installer uses the digest to pull the payload.

Add a new `release.txt` file to the release output that contains info about the release and a brief description indicating that the installer is locked. Add a pull spec to `oc adm release info` by default.

Add `--signing-key=PRIVATEKEY` to `extract --tools` to support simple signing of the sha256sum.txt file to create integrity on release. This support will remain simple and more complex use cases can be done with gpg.

The command:

```
oc adm release extract --tools --command-os=* --to=DIR IMAGE --signing-key=KEY
```

will:

1. Prompt to decrypt the private key in KEY
2. Download all tool into archives to DIR
3. Create `release.txt` containing similar output to `oc adm release info`
4. Create `sha256sum.txt` referencing all of the above
5. Sign `sha256sum.txt` with KEY as `sha256sum.txt.asc`

From this root you can then download the tar, verify it, install, trust the root, and be assured that the release you install is the release the tool extractor intended.  Subsequent updates can piggyback on this trust.

### CODE REVIEW

**Comment by smarterclayton** [comment]
@wking this is trying to provide just enough that we can use it for most users without getting into the weeds of gpg (they can do that external), going over user's heads, but still providing some basic info.

**Comment by wking** [comment]
[e2e-aws-serial][1]:

```
level=fatal msg="failed to initialize the cluster: Cluster operator console has not yet reported success: timed out waiting for the condition"
```

That was [rhbz#1699469[2].

/retest

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/22555/pull-ci-openshift-origin-master-e2e-aws-serial/4581
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=1699469

File: pkg/oc/cli/admin/release/extract_tools.go [hunk 1]
```diff
@@ -125,13 +132,50 @@ func (o *ExtractOptions) extractCommand(command string) error {
...
+			}
+			if err := key.PrivateKey.Decrypt(password); err != nil {
+				return fmt.Errorf("unable to decrypt signing key: %v", err)
+			}
+			for i, subkey := range key.Subkeys {
+				if err := subkey.PrivateKey.Decrypt(password); err != nil {
+					return fmt.Errorf("unable to decrypt signing subkey %d: %v", i, err)
+				}
+			}
+			signer = key
```
Comment (wking) [inline pkg/oc/cli/admin/release/extract_tools.go]: `break`?

Comment (wking) [reply pkg/oc/cli/admin/release/extract_tools.go]: Not having a `break` is not really a big deal, this UX isn't set up for keyfiles that contain multiple signing keys anyway, and it's probably not worth another round of CI ;).

File: pkg/oc/cli/admin/release/extract_tools.go [hunk 2]
```diff
@@ -339,10 +416,20 @@ func (o *ExtractOptions) extractCommand(command string) error {
 			hash := hashByTargetName[k]
 			lines = append(lines, fmt.Sprintf("%s  %s", hash, filepath.Base(k)))
 		}
-		filename := "sha256sum.txt.asc"
-		if err := ioutil.WriteFile(filepath.Join(dir, filename), []byte(strings.Join(lines, "\n")), 0644); err != nil {
+		data := []byte(strings.Join(lines, "\n"))
```
Comment (wking) [inline pkg/oc/cli/admin/release/extract_tools.go]: This is missing a trailing newline.

**Comment by wking** [comment]
Two nits, nothing serious, we can circle back to them later, maybe :p

/lgtm

**Comment by wking** [comment]
needs a rebase

**Comment by smarterclayton** [comment]
Rebased, reapplying label

**Comment by smarterclayton** [comment]
Blocked for too long, force merging so I can update the web pages

**Pull request merged**